lib/active_merchant/billing/gateways/sage_pay.rb

```diff
@@ -195,14 +215,11 @@ def add_customer_data(post, options)
       end
 
       def add_optional_data(post, options)
-        add_pair(post, :GiftAidPayment, options[:gift_aid_payment]) unless options[:gift_aid_payment].blank?
```

Comment:
The logic seems to have slightly changed here, previously it would add nothing if `option[:gift_aid_payment]` was `blank?`, now it looks like it will include `nil` for all keys which aren't present in `options`. Was this intentional / are we sure this is okay?